### PR TITLE
Added DisplayPromptAsync to macOS

### PR DESCRIPTION
**Pull request by amirvenus**
### Description of Change ###

-Added DisplayPromptAsync to macOS
-Added SubscribeToAlertsAndActionSheets for more code consistency across the macOS and iOS implementation

 
### API Changes ###
 
 
 None

### Platforms Affected ### 

- macOS

### Behavioral/Visual Changes ###

A user can now capture a value when using await DisplayPromptAsync on the macOS platform

 
### PR Checklist ###
 

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.MacOS/Platform.cs [hunk 1]
```diff
@@ -73,6 +108,21 @@ internal Platform()
...
+		{
+			_renderer = new PlatformRenderer(this);
+			_modals = new System.Collections.Generic.List<Page>();
+
+			SubscribeToAlertsAndActionSheets();
+
+
+		}
+
+		public void PresentPrompt(PromptArguments arguments)
```
Comment (VladislavAntonyuk) [inline Xamarin.Forms.Platform.MacOS/Platform.cs]: Please remove the empty method

File: Xamarin.Forms.Platform.MacOS/Platform.cs [hunk 2]
```diff
@@ -73,6 +108,21 @@ internal Platform()
 			});
 		}
 
+		internal Platform()
+		{
+			_renderer = new PlatformRenderer(this);
+			_modals = new System.Collections.Generic.List<Page>();
+
+			SubscribeToAlertsAndActionSheets();
+
```
Comment (VladislavAntonyuk) [inline Xamarin.Forms.Platform.MacOS/Platform.cs]: Cleanup code, remove empty lines

**Comment by amirvenus** [comment]
I have made the required changes

**Comment by amirvenus** [comment]
The UI tests are failing. 

How can I fix it?

**Comment by jfversluis** [comment]
Oh man! I would love to have this! But I tested it for a bit and it shows me some weird things. There is a little empty window behind the prompt and the prompt shows in a corner of the screen. And I don't really have time to spend on fixing this.

Thank you so much for your time and effort on this @amirvenus! Maybe you're able to port something for .NET MAUI if the team already hasn't?

**Pull request closed without merging**